consensus/cryptarchia-engine/src/lib.rs

```diff
@@ -320,4 +321,48 @@ pub mod tests {
         res[..8].copy_from_slice(&hash.to_be_bytes());
         res
     }
+
+    #[test]
+    fn test_getters() {
+        let engine = Cryptarchia::from_genesis([0; 32], config());
+        let parent = engine.genesis();
+
+        // Get branch directly from HashMap
+        let branch1 = engine
+            .branches
+            .get(&parent)
+            .ok_or("At least one branch should be there")
+            .expect("branch1 is not set");
+
+        let branches = engine.branches();
+
+        // Get branch using getter
+        let branch2 = branches
+            .get(&parent)
+            .ok_or("At least one branch should be there")
+            .expect("branch2 is not set");
+
+        assert_eq!(branch1, branch2);
+        assert_eq!(branch1.id(), branch2.id());
+        assert_eq!(branch1.parent(), branch2.parent());
+        assert_eq!(branch1.slot(), branch2.slot());
+        assert_eq!(branch1.length(), branch2.length());
+
+        let slot = Slot::genesis();
+
+        assert_eq!(slot + 10u64, Slot::from(10));
+
+        let strange_engine = Cryptarchia::from_genesis([100; 32], config());
+        let not_a_parent = strange_engine.genesis();
```

Comment:
that's quite a convoluted way to get an id, it should just be a `[u8; 32]`?

Reply:
You are right, fixed at https://github.com/logos-co/nomos-node/pull/657/commits/cbd68e6a16c55b4084522a1aa6bf81e37bfd1afc